src/corehost/cli/context_contract.h

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __CONTEXT_CONTRACT_H__
+#define __CONTEXT_CONTRACT_H__
+
+#include <pal.h>
+
+typedef void* context_handle;
+
+enum class coreclr_delegate_type
+{
+    invalid,
+    com_activation,
+    load_in_memory_assembly,
+    winrt_activation
+};
+
+struct corehost_context_contract
+{
+    size_t version;
+    context_handle instance;
+    int (*get_property_value)(
```

Comment:
Should we also declare the calling convention?
I think it would be safer... 
Aaron was asking that we explicitly declare calling convention everywhere... but I'm no expert on this topic.